File: src/main/java/com/fish/common/Result.java
package com.fish.common;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fish.exception.ServiceException;
import com.fish.exception.ServiceExceptionEnum;
import lombok.Data;

import java.io.Serializable;

@Data
@JsonInclude(Include.NON_NULL)
public class Result<T> implements Serializable {
    private int code;
    private String message;
    private T data;
    public Result(ServiceException exception) {
        this.data = null;
        this.code = exception.getCode();
        this.message = exception.getMsg();
    }
    public Result(ServiceExceptionEnum exceptionEnum) {
        this.data = null;
        this.code = exceptionEnum.getCode();
        this.message = exceptionEnum.getMsg();
    }
    public Result(int code, String message) {
        this.code = code;
        this.message = message;
        this.data = null;
    }
    public Result(int code, String message, T data) {
        this.code = code;
        this.message = message;
        this.data = data;
    }
    public Result(T data) {
        this.data = data;
        this.code = 200;
        this.message = "success";
    }
}

File: src/main/kotlin/com/fish/common/Result.kt
package com.fish.common

import com.fasterxml.jackson.annotation.JsonInclude
import com.fish.exception.ServiceException
import com.fish.exception.ServiceExceptionEnum
import java.io.Serializable

@JsonInclude(JsonInclude.Include.NON_NULL)
data class Result<T>(val code: Int,val message: String, val data: T?) : Serializable {
    constructor(exception: ServiceException) : this(exception.code, exception.msg, null)

    constructor(exceptionEnum: ServiceExceptionEnum) : this(exceptionEnum.code, exceptionEnum.msg, null)

    constructor(code: Int, message: String) : this(code, message, null)

    constructor(data: T) : this(200, "success", data)
}
